elf/elf_loader.c

```diff
@@ -51,10 +48,6 @@ void load_segment(uintptr_t base_addr, ELF_PHDR *phdr, int fd, Elf32_Half type,
   int prot = 0;
   uintptr_t aligned_vaddr, aligned_fsize, aligned_msize, page_offset, map_file_end;
 
-  /*if (phdr->p_flags & PF_X) {
-    prot |= PROT_EXEC;
-  }*/
-
```

Comment:
Don't remove this code, it sets the permissions needed to execute the loaded code directly. If you want, you can wrap it in an `#ifdef` and turn it off that way

Reply:
Done. Check to see if you agree with the label of the `#ifdef`.